pkg/assetfetcher/http_test.go

```diff
@@ -374,8 +361,9 @@ func TestFetchAssetsHTTPError(t *testing.T) {
 	defer ts.Close()
 
 	nexposeAssetFetcher := &NexposeAssetFetcher{
-		Host:     "fail://",
-		PageSize: 100,
+		HTTPClient: ts.Client(),
```

Comment:
This and a few other tests are still using `httptest.Server` where they should be using the mocked `RoundTripper`.

Reply:
Yeah, I left them in there because I couldn't figure out how cover those particular errors. I think I figured it out though. I'll make those changes. Thanks

Reply:
Ah, yes. I can use a mock for this test, but I can't figure out how to make `ioutil.ReadAll` fail to test that error. Any tips?

Reply:
Yeah, if you need to simulate a read error you can pass either gomock version of a reader or make a small struct to do the same thing. For example:

```golang
rt := NewMockRoundTripper(ctrl)
rd := NewMockReader(ctrl)

rt.EXPECT().RoundTrip(gomock.Any()).Return(
  &http.Response{Body: ioutil.NopCloser(rd)}, // and whatever other fields need setting.
  nil,
)
rd.EXPECT().Read(gomock.Any()).Return(0, errors.New("problem"))
// ... rest of the test
```
Or with a custom struct:

```golang
type errReader struct{
  Error error
}
func (r *errReader) Read(_ []byte) (int, error) {
  return 0, r.Error
}

func TestTheThing(t *testing.T) {
// other test stuff
rt := NewMockRoundTripper(ctrl)
rd := &errReader{Error: errors.New("problem")}

rt.EXPECT().RoundTrip(gomock.Any()).Return(
  &http.Response{Body: ioutil.NopCloser(rd)}, // and whatever other fields need setting.
  nil,
)
// ... rest of the test
```

Reply:
To piggyback on Kevin's suggestion, we have used the latter method before in Stride: https://bitbucket.org/hipchat/notifiable-store/src/9eedd84d46d8557cf47f731ff24442d1dadf9902/lib/handlers/query_test.go#lines-24

Reply:
Makes sense. Thanks y'all, I'll add it in.